### PR TITLE
Restricted Python, but ~~butter~~ better

### PR DESCRIPTION
**Pull request by analogrelay**
Fixes aseriousbiz/abbot#3765

This restores the Restricted Python-based sandboxing to full strength, with a few changes:

* More Python features work. I tested list comprehensions and sequence unpacking. I'll go to some of our existing skills that are failing with the current sandboxing and see if I can make sure they're working now.
* Less restrictive module loading. We now have two lists: Allow and Deny. If a module is in Deny, it and all child modules are forbidden from being loaded. If a module is in Allow, it and all child modules may be loaded without warning. Otherwise, a warning is logged so we can track and investigate the safety of the module.

TODO:
* [x] Validate in canary
* [x] Check existing python skills to confirm they'll work now.

### CODE REVIEW

**Comment by analogrelay** [comment]
Looks like this should resolve the issues with existing Python skills

File: src/SkillRunner/bot/policy/restrictive.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
+"""
+Defines the RestrictedPolicy, which uses RestrictedPython to strictly limit what the script can do.
```
Comment (haacked) [inline src/SkillRunner/bot/policy/restrictive.py:2]:
```suggestion
Defines the RestrictivePolicy, which uses RestrictedPython to strictly limit what the script can do.
```

**Review by haacked — APPROVED** [review]
🚢 Nice to have tests of the policies now. Nice work!

File: src/SkillRunner/bot/policy/__init__.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    "stat", # file I/O
+    "struct", # binary serialization
+    "subprocess", # process management
+    "symtable", # compiler internals
+    "sys",
+    "sysconfig",
+    "syslog", # I/O
+    "threading", # threading
+    "venv", # no development-time tomfoolery
+   	"warnings", # Why would they need to do this?
```
Comment (dahlbyk) [inline src/SkillRunner/bot/policy/__init__.py:73]:
```suggestion
    "warnings", # Why would they need to do this?
```

File: src/SkillRunner/bot/policy/restrictive.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    Executing code in this environment will call back to the delegate
+    to implement the built-in functionality.
+    """
+    def __init__(self, logger: logging.Logger, delegate):
+        # Restricted Python works by translating the incoming Python code
+        # into calls to well-known functions.
+        # Then, we provide implementations of those functions as globals.
+        # For example, `import foo` becomes `__import__("foo")`, and `foo.bar`
+
+        # For completeness, we
```
Comment (dahlbyk) [inline src/SkillRunner/bot/policy/restrictive.py:201]:
```suggestion
```

**Review by dahlbyk — APPROVED** [review]

**Review by haacked — APPROVED** [review]
:shipit:

**Pull request merged**